File: frontend/src/types/BookDto.ts
import {FileResourceDto} from "./FileResourceDto";
import {UserDto} from "./UserDto";
import {BookType} from "./BookType";

export interface BookDto {
    id: number,
    name: string,
    description: string,
    bookType: BookType,
    fileResource: FileResourceDto,
    image: FileResourceDto,
    createdBy: UserDto
}

File: src/main/java/com/ivank/bookup/dto/BookDto.java
package com.ivank.bookup.dto;

import com.ivank.bookup.model.enums.BookType;
import lombok.Data;

@Data
public class BookDto {

    public Long id;

    public String name;

    public String description;

    public BookType type;

    public FileResourceDto fileResource;

    public UserDto createdBy;

    public FileResourceDto image;

}
